packages/material-ui/src/Popper/Popper.js

```diff
@@ -187,15 +190,22 @@ Popper.propTypes = {
           return new Error(
             [
               'Material-UI: the `anchorEl` prop provided to the component is invalid.',
-              'The node element should be visible.',
+              'The reference element should have a sound position in the page.',
```

Comment:
Hmm, I'm wondering if we should swap 'sound' for another adjective.
e.g. reasonable, logical, etc.
```suggestion
              'The reference element should have a sound position on the page.',
```

Reply:
@joshwooding The reference element should be mounted and displayed?

Reply:
> displayed: (of information) shown on a computer screen or other device.

Does it cover `visibility: hidden`?

"The reference element should be part of the layout of on the page"?

Reply:
@oliviertassinari  :+1: "The reference element should be part of the **document layout**"

Reply:
Hmm 🤔 Technically "The reference element should be part of the layout on the page" is correct but requires prior knowledge to understand. It seems like we would get a lot of questions about it.

Reply:
I believe we can solve this problem by proposing different solutions. I have added `It might no longer be mounted or display none.`. But we can further improve it. Maybe:
`Make sure the element is still present in the document or that it's not display none.`?

Reply:
Can we take a step back and rethink what cases we're trying to catch? Is this even an issue?

Reply:
@eps1lon It was introduced to solve these problems:
- #15009 (unmounted element)
- A use case I had in Onepixel where the tooltip was positioned at the top of the page for no obvious reason (ref forwarding issue)
- https://github.com/twbs/bootstrap/issues/14155 (help user understand that it doesn't work with display none)
- A case you have faced working on the "Show source" tooltip of the documentation.

Reply:
> A case you have faced working on the "Show source" tooltip of the documentation.

What PR was this? Can't remember getting a warning

Reply:
It was before we introduced the warning, I can't find the history. Ignore my last point.

Reply:
Didn't know display: none set the position to 0,0 makes sense in hindsight.